firefox-ios/Client/ApplicationHelper.swift

```diff
@@ -46,4 +47,18 @@ struct DefaultApplicationHelper: ApplicationHelper {
             open(url)
         }
     }
+
+    func closeTab(_ url: URL) async {
+        // We need to get all the windows/tabs and then find the url that was requested to close
+        await UIApplication.shared.connectedScenes.forEach { scene in
+            let windowManager = AppContainer.shared.resolve() as WindowManager
+            windowManager.allWindowTabManagers().forEach { tabManager in
+                if let tabManagerImplementation = tabManager as? TabManagerImplementation {
```

Comment:
Ideally I don't think we should be typecasting to `TabManagerImplementation` here. I would suggest that we add the new `closeTab` function to the `TabManager` protocol (though might be good to get additional eyes on this from one of our tab experts cc @OrlaM @nbhasin2 ).

Reply:
Great suggestion here -- removed all the unnecessary casting and simplified the call-sites. Hopefully looks cleaner now!